### PR TITLE
feat: add plugin default config

### PR DESCRIPTION
**Pull request by steinliber**
Signed-off-by: Meng JiaFeng <jiafeng.meng@merico.dev>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description
- All helm plugins support the default value 

## Related Issues
#523 #526 #527 #513

## New Behavior (screenshots if needed)
Jenkins config Before
image

After
image

### CODE REVIEW

File: docs/plugins/artifactory.md [hunk 1]
```diff
@@ -44,4 +44,15 @@ This plugin support`Ingress`, `ClusterIP`, `NodePort` and `LoadBalancer` , You c
...
+| key | default value | description |
+|  ----  | ----  | ----  |
+| chart.chart_name | jfrog/artifactory | chart name |
+| chart.timeout | 10m | this config will wait 10 minutes to deploy |
+| upgradeCRDs | true | default update CRD config |
+| chart.wait | true | whether to wait until install is complete |
+| repo.url | https://charts.jfrog.io | offical helm repo address |
+| repo.name | jfrog | helm repo name |
+
+Currently, except for `values_yaml` and default configs, all the parameters in the example above are mandatory.
```
Comment (iyear) [inline docs/plugins/artifactory.md:58]: I see this sentence, but there is no action in the code to set a default value to `values_yaml`.

It is also possible that I missed it.

File: docs/plugins/jenkins.md [hunk 2]
```diff
@@ -46,7 +46,19 @@ chown -R 1000:1000 <your-dtm-home-dir>/data/jenkins-volume/
 --8<-- "jenkins.yaml"
 ```
 
-Currently, all the parameters in the example above are mandatory.
+#### Default Configs
+
+| key | default value | description |
+|  ----  | ----  | ----  |
+| chart.chart_name | jenkins/jenkins | chart name |
+| chart.timeout | 10m | this config will wait 10 minutes to deploy |
```
Comment (iyear) [inline docs/plugins/jenkins.md:54]: The default value in the yaml file is 5 minutes, but the code is set to 10 minutes.

File: docs/plugins/jenkins.zh.md [hunk 3]
```diff
@@ -46,7 +46,18 @@ chown -R 1000:1000 <your-dtm-home-dir>/data/jenkins-volume/
 --8<-- "jenkins.yaml"
 ```
 
-当前，所有配置项均为必填。
+#### 默认配置
+
+| key | default value | description |
+|  ----  | ----  | ----  |
+| chart.chart_name | jenkins/jenkins | helm 包名称 |
+| chart.timeout | 10m | 等待部署成功的时间 |
```
Comment (iyear) [inline docs/plugins/jenkins.zh.md:54]: The default value in the yaml file is 5 minutes, but the code is set to 10 minutes.

File: docs/plugins/kube-prometheus.md [hunk 4]
```diff
@@ -8,4 +8,15 @@ This plugin installs [kube-prometheus](https://github.com/prometheus-operator/ku
 --8<-- "kube-prometheus.yaml"
 ```
 
-Currently, except for `values_yaml`, all the parameters in the example above are mandatory.
+### Default Configs
+
+| key | default value | description |
+|  ----  | ----  | ----  |
+| chart.chart_name | prometheus-community/kube-prometheus-stack | chart name |
+| chart.timeout | 10m | this config will wait 10 minutes to deploy |
```
Comment (iyear) [inline docs/plugins/kube-prometheus.md:16]: The default value in the yaml file is 5 minutes, but it is set to 10 minutes in the code and default config.

**Comment by iyear** [comment]
I have compared the original yaml, the code `defaultHelmConfig` and the default config table in the doc.

**Review by iyear — APPROVED** [review]
More user-friendly!

File: docs/plugins/argocd.md [hunk 5]
```diff
@@ -10,13 +10,13 @@ This plugin installs [ArgoCD](https://argoproj.github.io/cd/) in an existing Kub
...
-| repo.url | https://argoproj.github.io/argo-helm | helm repo address |
-| repo.name | argo | helm repo name |
+|       key        | default value                        | description                                      |
+|       ----       | ----                                 | ----                                             |
+| chart.chart_name | argo/argo-cd                         | chart name                                       |
+|  chart.timeout   | 5m                                   | this config will wait 5 minutes to deploy argocd |
+|   upgradeCRDs    | true                                 | default update CRD config                        |
+|    chart.wait    | true                                 | whether to wait until install is complete        |
+|     repo.url     | https://argoproj.github.io/argo-helm | helm official repo address                       |
+|    repo.name     | argo                                 | helm repo name                                   |
```
Comment (aFlyBird0) [inline docs/plugins/argocd.md:20]: I see that in the first column of your table, some of the documents are center-aligned like this, and some of them are left-aligned.

Comment (aFlyBird0) [reply docs/plugins/argocd.md:20]: After testing, I found that if you edit the file a bit (e.g. delete a space in the table and add it back), then Ctrl+S, Goland will automatically align it to the left for you. Try it?

File: docs/plugins/harbor.md [hunk 6]
```diff
@@ -68,4 +68,15 @@ This plugin support `Ingress`, `ClusterIP`, `NodePort`, `LoadBalancer` , You can
 --8<-- "harbor.yaml"
 ```
 
-Currently, except for `values_yaml`, all the parameters in the example above are mandatory.
+#### Default Configs
+
+| key              | default value            | description                                |
+| ----             | ----                     | ----                                       |
+| chart.chart_name | harbor/harbor            | chart name                                 |
+| chart.timeout    | 10m                      | this config will wait 10 minutes to deploy |
```
Comment (aFlyBird0) [inline docs/plugins/harbor.md:76]: image

Please edit this comment.

File: docs/plugins/argocd.md (hunk 5 above)
Comment (aFlyBird0) [reply docs/plugins/argocd.md:20]: > it's editor's operation， which one looks better

I prefer the left-aligned one?

File: docs/plugins/artifactory.zh.md [hunk 7]
```diff
@@ -44,4 +44,15 @@ values_yaml: |
 --8<-- "artifactory.yaml"
 ```
 
-目前除了 `values_yaml` 字段，所有示例参数均为必填项。
+#### 默认配置
+
+| key              | default value           | description                       |
+| ----             | ----                    | ----                              |
+| chart.chart_name | jfrog/artifactory       | helm 包名称                       |
+| chart.timeout    | 10m                     | 等待部署成功的时间                |
```
Comment (aFlyBird0) [inline docs/plugins/artifactory.zh.md:52]: image

File: docs/plugins/openldap.md [hunk 8]
```diff
@@ -8,6 +8,17 @@ This plugin installs [OpenLDAP](https://www.openldap.org/) in an existing Kubern
 --8<-- "openldap.yaml"
 ```
 
+### Default Configs
+
+|       key        | default value                             | description                                |
+|       ----       | ----                                      | ----                                       |
+| chart.chart_name | helm-openldap/openldap-stack-ha           | community chart name                       |
+|  chart.timeout   | 5m                                        | this config will wait 10 minutes to deploy |
```
Comment (aFlyBird0) [inline docs/plugins/openldap.md:16]: "5m"   vs "this config will wait 10 minutes to deploy"

File: docs/plugins/hashicorp-vault.md [hunk 9]
```diff
@@ -10,6 +10,17 @@ This plugin installs hashicorp-vault with replicas:3 by default value.
 --8<-- "hashicorp-vault.yaml"
 ```
 
+### Default Configs
+
+| key              | default value                       | description                                |
+| ----             | ----                                | ----                                       |
+| chart.chart_name | hashicorp/vault                     | chart name                                 |
+| chart.timeout    | 5m                                  | this config will wait 10 minutes to deploy |
```
Comment (aFlyBird0) [inline docs/plugins/hashicorp-vault.md:18]: "5m" vs "this config will wait 10 minutes to deploy"

File: docs/plugins/hashicorp-vault.md [hunk 10]
```diff
@@ -10,6 +10,17 @@ This plugin installs hashicorp-vault with replicas:3 by default value.
...
+| upgradeCRDs      | true                                | default update CRD config                  |
+| chart.wait       | true                                | whether to wait until install is complete  |
```
Comment (aFlyBird0) [inline docs/plugins/hashicorp-vault.md:20]: "install" is a verb.

image

You could use "whether to wait until the installation is complete"

**Review by aFlyBird0 — CHANGES_REQUESTED** [review]

**Comment by aFlyBird0** [comment]
A very helpful documentation improvement for users!👏

**Review by aFlyBird0 — APPROVED** [review]

**Comment by daniel-hutao** [comment]
@steinliber Please resolve the conflicts. Then we can merge this pr asap.

**Review by IronCore864 — APPROVED** [review]

**Pull request closed without merging**